### ISSUE TITLE
Make sure kopia passes all tests on arm and arm64 (was: Panic/segfault when creating snapshot on arm)

### ISSUE THREAD

**Issue by aksdb** [kopia/kopia#505]
* The repository in question was created with 0.5.2.
* Snapshot is being created using kopia 0.6-rc1
* Platform is **arm**.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11f08]

goroutine 1356 [running]:
runtime/internal/atomic.goStore64(0x70a6914, 0xe95a467c, 0x1625f609)
        /home/travis/.gimme/versions/go1.14.6.linux.amd64/src/runtime/internal/atomic/atomic_arm.go:144 +0x1c
github.com/kopia/kopia/repo/content.(*cacheBase).perItemMutex(0x2df61e0, 0x9a1068, 0x2f4a310, 0x0)
        /home/travis/gopath/src/github.com/kopia/kopia/repo/content/content_cache_base.go:68 +0x120
github.com/kopia/kopia/repo/content.(*contentCacheForMetadata).getContent(0x2cb2560, 0xc93ef0, 0x304e120, 0x2cd0120, 0x21, 0x2cd0120, 0x21, 0x0, 0x0, 0xffffffff, ...)
        /home/travis/gopath/src/github.com/kopia/kopia/repo/content/content_cache_metadata.go:53 +0x74
github.com/kopia/kopia/repo/content.(*indexBlobManagerImpl).getEncryptedBlob(0x2ca6440, 0xc93ef0, 0x304e120, 0x2cd0120, 0x21, 0x3ac578, 0x2c647c0, 0x2e45fbc, 0x1, 0x101)
        /home/travis/gopath/src/github.com/kopia/kopia/repo/content/index_blob_manager.go:145 +0x70
github.com/kopia/kopia/repo/content.(*indexBlobManagerImpl).getIndexBlob(0x2ca6440, 0xc93ef0, 0x304e120, 0x2cd0120, 0x21, 0x0, 0x0, 0x2d93c40, 0x2d93c40, 0x0)
        /home/travis/gopath/src/github.com/kopia/kopia/repo/content/index_blob_manager.go:141 +0x3c
github.com/kopia/kopia/repo/content.(*lockFreeManager).tryLoadPackIndexBlobsUnlocked.func1(0x2e706b0, 0x2c647c0, 0x2c7e044, 0xc93ef0, 0x304e120, 0x2c64840)
        /home/travis/gopath/src/github.com/kopia/kopia/repo/content/content_manager_lock_free.go:162 +0xc8
created by github.com/kopia/kopia/repo/content.(*lockFreeManager).tryLoadPackIndexBlobsUnlocked
        /home/travis/gopath/src/github.com/kopia/kopia/repo/content/content_manager_lock_free.go:158 +0x204
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11f08]

goroutine 1352 [running]:
runtime/internal/atomic.goStore64(0x7118004, 0xe95a467c, 0x1625f609)
        /home/travis/.gimme/versions/go1.14.6.linux.amd64/src/runtime/internal/atomic/atomic_arm.go:144 +0x1c
github.com/kopia/kopia/repo/content.(*cacheBase).perItemMutex(0x2df61e0, 0x9a1068, 0x2d1c160, 0x0)
        /home/travis/gopath/src/github.com/kopia/kopia/repo/content/content_cache_base.go:68 +0x120
github.com/kopia/kopia/repo/content.(*contentCacheForMetadata).getContent(0x2cb2560, 0xc93ef0, 0x304e120, 0x2cd40c0, 0x21, 0x2cd40c0, 0x21, 0x0, 0x0, 0xffffffff, ...)
        /home/travis/gopath/src/github.com/kopia/kopia/repo/content/content_cache_metadata.go:53 +0x74
github.com/kopia/kopia/repo/content.(*indexBlobManagerImpl).getEncryptedBlob(0x2ca6440, 0xc93ef0, 0x304e120, 0x2cd40c0, 0x21, 0x3ac578, 0x2c647c0, 0x2e40fbc, 0x1, 0x101)
        /home/travis/gopath/src/github.com/kopia/kopia/repo/content/index_blob_manager.go:145 +0x70
github.com/kopia/kopia/repo/content.(*indexBlobManagerImpl).getIndexBlob(0x2ca6440, 0xc93ef0, 0x304e120, 0x2cd40c0, 0x21, 0x2e40fcc, 0x1, 0x341828, 0x1, 0x74d44)
        /home/travis/gopath/src/github.com/kopia/kopia/repo/content/index_blob_manager.go:141 +0x3c
github.com/kopia/kopia/repo/content.(*lockFreeManager).tryLoadPackIndexBlobsUnlocked.func1(0x2e706b0, 0x2c647c0, 0x2c7e044, 0xc93ef0, 0x304e120, 0x2c64840)
        /home/travis/gopath/src/github.com/kopia/kopia/repo/content/content_manager_lock_free.go:162 +0xc8
created by github.com/kopia/kopia/repo/content.(*lockFreeManager).tryLoadPackIndexBlobsUnlocked
        /home/travis/gopath/src/github.com/kopia/kopia/repo/content/content_manager_lock_free.go:158 +0x204
```

**Comment by jkowalski** [comment]
Thanks for the report. We don't have CI/CD coverage yet for arm and arm64 unfortunately (other than cross-compiling). I used my Raspberry Pi at home and was able to reproduce this. I think we should try to get the build to a state where it can natively build on arm/linux and arm64/linux and pass at least unit tests. I'm going to mark this bug for 0.6.

**Comment by jkowalski** [comment]
Travis CI appears to have some support for ARM.

**Comment by jkowalski** [comment]
Tried running tests on aarm64:

Linux 5.4.0-1013-raspi #13-Ubuntu SMP Mon Jun 15 03:17:37 UTC 2020 aarch64 aarch64 aarch64 GNU/Linux

E2E tests all passed and almost all unit tests too, the only one that failed was due to build failure on fswalker lib.

**Comment by aksdb** [comment]
Btw I think the underlying problem is this: https://golang.org/pkg/sync/atomic/#pkg-note-BUG

I was not aware of that restriction until now.

**Comment by jkowalski** [comment]
The fixes are in 0.6.0-rc2 release
